### ISSUE TITLE
Error reporting is forcibly disabled

### ISSUE THREAD

**Issue by thany** [zenphoto/zenphoto#1218]
First of all, PHP error reporting is a server setting and should be respected as such.

Zenphoto forcibly sets `display_errors` to `off`, which is why I can't see what errors I'm causing while developing my theme. I need to see those errors. Currently, *any* PHP error causes it to basically `die()` and leave me with half a page and none the wiser as to what's going on.

In `functions-basic.php`:
```php
@ini_set('display_errors', '0'); // try to disable in case set
if (TEST_RELEASE) {
	error_reporting(E_ALL | E_STRICT);
	@ini_set('display_errors', '1');
} 
```
The `TEST_RELEASE` constant is falsey so `display_errors` will remain disabled. This setting will override the server setting in php.ini, .user.ini and .htaccess. To make `TEST_RELEASE` truthy, it needs to be running a special version of zenphoto, it seems (as a matter of fact, I'm running master) - and this is probably a good choice for internal zenphoto errors.

But as for **theme** errors, we'd most likely want to still see them, especially if the server excplicitly tells php to enable error reporting! Now I don't think there's a way to distinguish between theme errors and internal zenphoto errors, so better to enable all of them.

To recap:
1) Why is it done like this??
2) How to override without changing anything in zp-core?

**Comment by acrylian** [comment]
> Why is it done like this??

Generally display errors should be off. Remember we aim not only at developers but "normal users" and some servers have this on for weird reasons.

Zenphoto obviously has its own debug log that will tell you about any error while developing (located in `zp-data`. Additionally your server should have a PHP error log. How and if that is accessible your host will tell you.

**Comment by thany** [comment]
I don't think it is a good idea to put weird things in zenphoto, just because some servers decide it is a good idea to turn on error reporting. Fix those servers, because the software is fine. Better yet, instruct folks to put the corresponding lines in their .htaccess (or whatever the equivalent is for their server).

An error log is pretty good and all, but it lacks the immediacy of `display_errors`. Just hit F5 and boom: there's your error, ready to get solved.

The error log on my server doesn't seem to be logging anything but fatal errors, for some reason, even though I'm *displaying* all errors, warnings, and notices. Perhaps it's a quirk of XAMPP. Either way, I prefer it if they are rendered into the page.

**Comment by acrylian** [comment]
Try the plugin to enable error display if you want it. I personally never use it because I find logs more convenient. Each his own.

>Besides: is this still actually an issue, should you remove this? Iow, are there actually really still production servers in current use that have display_errors on?

I wonder as well but I have just recently encounterd that. And not every user is savy to edit a htaccess file or even knows what it is. We seem to have a lot users that use install scripts to install as well.

**Comment by acrylian** [comment]
Addition: But there are plans to move a lot debug defines currently in core files to the config file so you can more easily enable them if you want them and know what you are doing.
